Support `--frozen` option for routing annotations

kg8m opened the pull request:
Hi. Thanks for this great gem! The `--frozen` option is very useful.

I noticed that the `--frozen` option doesn't work for routing annotations (`rake annotate_routes` and `annotate --routes`). This pull request adds the support for that.

Best regards.

kg8m committed:
Support `--frozen` option for routing annotations

Signed-off-by: kg8m <takumi.kagiyama@gmail.com>

ctran commented:
@kg8m Would you mind rebase against the latest `develop` (which has fixed the CI issues)?

Thanks.

ctran commented on lib/annotate/annotate_routes.rb:
```diff
@@ -30,10 +30,18 @@ def do_annotations(options = {})
+            puts "#{routes_file} was annotated."
+          else
+            puts "#{routes_file} was not changed."
+          end
```
Rubocop wants to convert this into an `elsif`.

ctran commented:
Could I convince you to add an unit test/spec for this new option?

kg8m committed:
Prefer `elsif` to nested `if`s

Signed-off-by: kg8m <takumi.kagiyama@gmail.com>

kg8m committed:
Add tests for `frozen` option of routing annotations

Signed-off-by: kg8m <takumi.kagiyama@gmail.com>

kg8m commented:
@ctran 
I fixed the RuboCop error and added tests for the `--frozen` option of routing annotations. Could you please review the changes?

Best regards.

ctran commented on lib/annotate/annotate_routes.rb:
```diff
@@ -30,7 +30,13 @@ def do_annotations(options = {})
+        elsif rewrite_contents(existing_text, new_text)
           puts "#{routes_file} was annotated."
         else
           puts "#{routes_file} was not changed."
```
I feel this can be somewhat improved to remove the code duplication and nested ifs.  What about something like this?

```ruby
# don't feel like a separate function is needed
content_changed = (existing_text.strip != new_text.strip)  

if !content_changed # you can flip this condition around if you want
  puts "#{routes_file} was not changed."
else
  abort "annotate error. #{routes_file} needs to be updated, but annotate was run with `--frozen`." if options[:frozen]
  rewrite_contents(existing_text, new_text)  
end
```

ctran replied:
Now that I'm looking at the code closer, maybe we should move all this inside the `rewrite_contents` function.

ctran replied:
This could be improved to remove code duplication and nested ifs:

```ruby
if rewrite_contents(existing_text, new_text)
  puts "#{routes_file} was annotated."
else
  puts "#{routes_file} was not changed."
end
```

Then change `#rewrite_contents` to:
```ruby
def rewrite_contents(existing_text, new_text)
  content_changed = (existing_text != new_text)

  # Rubocop may prefer "abort ... if"
  if content_changed && options[:frozen] 
    abort "annotate error. #{routes_file} needs to be updated, but annotate was run with `--frozen`." 
  end

  File.open(routes_file, 'wb') { |f| f.puts(new_text) }
  content_changed
end
```

ctran commented on spec/lib/annotate/annotate_routes_spec.rb:
Thanks for the test!

ctran committed:
Update annotate_routes.rb

ctran committed:
Update annotate_routes.rb

ctran committed:
Update annotate_routes.rb

ctran committed:
Update annotate_routes.rb

ctran reviewed (approved):
I made the suggestions directly on your fork, hope you don't mind.

kg8m commented:
@ctran 
Thank you so much!
